debugoverlay-ext-netstats/src/main/java/com/ms_square/debugoverlay_ext_netstats/NetStatsDataModule.java

```diff
@@ -53,10 +53,12 @@ public void stop() {
         public void run() {
             double totalBytesReceived = TrafficStats.getUidRxBytes(uid);
             double totalBytesSent = TrafficStats.getUidTxBytes(uid);
-            if (previousReceived == null)
+
+            if (previousReceived == TrafficStats.UNSUPPORTED || previousSent == TrafficStats.UNSUPPORTED) {
```

Comment:
I think we should not use the TrafficStats.UNSUPPORTED to represent the un-initialized state which has a different meaning. Also, the Android team could technically change it anytime to whatever value they want; thus, we should use it for only its intended purpose.

I would probably just do something like

```
private double previousReceived = -1f;
private double previousSent = -1f;

....
....
// within run()
double totalBytesReceived = TrafficStats.getUidRxBytes(uid);
double totalBytesSent = TrafficStats.getUidTxBytes(uid);

if (totalBytesReceived == TrafficStats.UNSUPPORTED || totalBytesSent == TrafficStats.UNSUPPORTED) {
    Log.w(TAG, "The use of TrafficStats is not supported on this device.");
    return;
}

if (previousReceived >= 0 && previousSent >= 0) {
    received = (totalBytesReceived - previousReceived) / intervalSeconds;
    sent = (totalBytesSent - previousSent) / intervalSeconds;
    notifyObservers();
    handler.postDelayed(networkStatisticsQueryRunnable, intervalMilliseconds);
}
previousReceived = totalBytesReceived;
previousSent = totalBytesSent;
```

Reply:
I like that! It's great that the ``Runnable`` returns without scheduling a new run if any of the values are ``TrafficStats.UNSUPPORTED``, that way the ``Runnable`` will only be called once and the error message will also just be shown once.
However the ``handler.postDelayed`` should be outside of the ``if`` as otherwise the ``Runnable`` will only run once.
I've also decided to move the "initialization" of the ``previous*`` variables into ``start`` since these variables should be reset if the module have been stopped.